sdk/keyvault/Microsoft.Azure.KeyVault/tests/ChallengeCacheHandlerTests.cs

```diff
@@ -0,0 +1,254 @@
+
+            AssertChallengeCacheEntry(requestUrl, expChallenge);
+        }
+
+        [Fact]
+        public async Task CacheUpdateOn401Async()
+        {
+            string requestUrlBase = CreateMockUrl();
+
+            string requestUrl1 = CreateMockUrl(requestUrlBase, 2);
+
+            HttpBearerChallengeCache.GetInstance().SetChallengeForURL(new Uri(requestUrl1), MockChallenge.Create().ToHttpBearerChallenge(requestUrl1));
+
+            string requestUrl2 = CreateMockUrl(requestUrlBase, 2);
+
+            var handler = new ChallengeCacheHandler();
+
+            var expChallenge = MockChallenge.Create();
+
+            handler.InnerHandler = new StaticChallengeResponseHandler(HttpStatusCode.Unauthorized, expChallenge.ToString());
+
+            var client = new HttpClient(handler);
+
+            var _ = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, requestUrl2));
+
+            AssertChallengeCacheEntry(requestUrl1, expChallenge);
+
+            AssertChallengeCacheEntry(requestUrl2, expChallenge);
+        }
+
+        [Fact]
+        public async Task CacheNotUpdatedNoChallengeAsync()
+        {
+            var handler = new ChallengeCacheHandler();
+
+            handler.InnerHandler = new StaticChallengeResponseHandler(HttpStatusCode.Unauthorized, null);
+
+            var client = new HttpClient(handler);
+
+            var requestUrl = CreateMockUrl(2);
+
+            var _ = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, requestUrl));
+
+            AssertChallengeCacheEntry(requestUrl, null);
+        }
+
+        [Fact]
+        public async Task CacheNotUpdatedNon401Aysnc()
+        {
+            var handler = new ChallengeCacheHandler();
+
+            var expChallenge = MockChallenge.Create();
+
+            handler.InnerHandler = new StaticChallengeResponseHandler(HttpStatusCode.Forbidden, expChallenge.ToString());
+
+            var client = new HttpClient(handler);
+
+            var requestUrl = CreateMockUrl(2);
+
+            var _ = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, requestUrl));
+
+            AssertChallengeCacheEntry(requestUrl, null);
+        }
+
+        [Fact]
+        public async Task CacheNotUpdatedNonBearerChallengeAsync()
+        {
+            var handler = new ChallengeCacheHandler();
+
+            handler.InnerHandler = new StaticChallengeResponseHandler(HttpStatusCode.Unauthorized, MockChallenge.Create("PoP").ToString());
+
+            var client = new HttpClient(handler);
+
+            var requestUrl = CreateMockUrl(2);
+
+            var _ = await client.SendAsync(new HttpRequestMessage(HttpMethod.Get, requestUrl));
+
+            AssertChallengeCacheEntry(requestUrl, null);
+        }
+
+        private static void AssertChallengeCacheEntry(string requestUrl, MockChallenge expChallenge)
+        {
+            var actChallenge = HttpBearerChallengeCache.GetInstance().GetChallengeForURL(new Uri(requestUrl));
```

Comment:
Are there any other tests these tests might conflict with when running in parallel?

Reply:
These shouldn't interfere with other tests as all the URLs that I'm using are guid based.